test/e2e/workflow/deployment_test.go

```diff
@@ -160,6 +161,50 @@ var _ = Describe("NetworkAddonsConfig", func() {
 			CheckConfigCondition(ConditionAvailable, ConditionTrue, 10*time.Minute, 30*time.Second)
 		})
 	})
+
+	Context("testing range generation after kubeMacPool reinstalation ", func() {
+		var (
+			config                                           opv1alpha1.NetworkAddonsConfigSpec
+			oldRangeEnd, oldRangeStart, rangeEnd, rangeStart string
+		)
+
+		BeforeEach(func() {
+			config = opv1alpha1.NetworkAddonsConfigSpec{KubeMacPool: &opv1alpha1.KubeMacPool{}}
+			CreateConfig(config)
+		})
+
+		AfterEach(func() {
+			DeleteConfig()
+			CheckComponentsRemoval(AllComponents)
+		})
+
+		When("installing kubeMacPool", func() {
+			BeforeEach(func() {
+				rangeStart, rangeEnd = CheckUnicastAndValidity()
+			})
+
+			It("should create kubeMacPool, generate a non-multicast valid range for it and deleting it afterwards", func() {
+				Expect(rangeStart).ToNot(Equal(rangeEnd))
+			})
+		})
+
+		When("reinstalling kubeMacPool", func() {
+			BeforeEach(func() {
+				oldRangeStart, oldRangeEnd = CheckUnicastAndValidity()
```

Comment:
comment before each with

By("Reinstalling KubeMacPool"), so there is easier to track what is going on in the log